src/storage/src/source/source_reader_pipeline.rs

```diff
@@ -733,7 +785,93 @@ where
         }
     });
 
-    ((batch_stream, summary_stream), Some(capability))
+    (
+        SourceReaderStreams {
+            batches: batch_stream,
+            batch_upper_summaries: summary_stream,
+            health_stream,
+        },
+        Some(capability),
+    )
+}
+
+/// Mints new contents for the remap shard based on summaries about the source
+/// upper it receives from the raw reader operators.
+///
+/// Only one worker will be active and write to the remap shard. All source
+/// upper summaries will be exchanged to it.
+fn health_operator<G>(
+    scope: &G,
+    config: RawSourceCreationConfig,
+    health_stream: timely::dataflow::Stream<G, (usize, HealthStatus)>,
+) -> Rc<dyn Any>
+where
+    G: Scope<Timestamp = Timestamp>,
+{
+    let RawSourceCreationConfig {
+        worker_id: healthcheck_worker_id,
+        worker_count,
+        id: source_id,
+        ..
+    } = config;
+
+    // We'll route all the work to a single arbitrary worker;
+    // there's not much to do, and we need a global view.
+    let chosen_worker_id = source_id.hashed() as usize % worker_count;
+    let is_active_worker = chosen_worker_id == healthcheck_worker_id;
+
+    let mut healths = vec![HealthStatus::Starting; worker_count];
+
+    let operator_name = format!("healthcheck({})", healthcheck_worker_id);
+    let mut health_op = OperatorBuilder::new(operator_name, scope.clone());
+
+    let mut input = health_op.new_input_connection(
```

Comment:
super nit: Just use `health_op.new_input`?  They're effectively the same for this operator with no outputs and the default would like be more correct were we to add an output.

Unless you're adding this as intentional future-proofing so we explicitly have to think about whether we should use the default or the empty antichain.  In which case: sounds great to me and maybe add a comment?